src/codegen/codegen_cuda_visitor.cpp

```diff
@@ -71,6 +74,10 @@ void CodegenCudaVisitor::print_atomic_op(const std::string& lhs,
 
 void CodegenCudaVisitor::print_backend_includes() {
     printer->add_line("#include <cuda.h>");
+
+    if (info.eigen_linear_solver_exist && count_length(info.state_vars) > 4) {
```

Comment:
not blocking but a one-liner with accumulate here is better than a new static function since the code remains cleaner. As you prefer